docs/src/manual/ml.md

```diff
@@ -0,0 +1,132 @@
+```@meta
+CurrentModule = JuliaDB
+DocTestSetup = quote
+    using JuliaDB
+    import JuliaDB: ML
+end
+```
+# Feature Extraction
+
+Machine learning models are composed of mathematical operations on matrices of numbers. However, data in the real world is often in tabular form containing more than just numbers. Hence, the first step in applying machine learning is to turn such tabular non-numeric data into a matrix of numbers. Such matrices are called "feature matrices". JuliaDB contains an `ML` module which has helper functions to extract feature matrices.
+
+In this document, we will turn the [titanic dataset from Kaggle](https://www.kaggle.com/c/titanic) into numeric form and apply a machine learning model on it.
+
+```@example titanic
+using JuliaDB
+
+download("https://raw.githubusercontent.com/agconti/"*
+          "kaggle-titanic/master/data/train.csv", "train.csv")
+
+train_table = loadtable("train.csv", escapechar='"')
+popcol(popcol(popcol(train_table, :Name), :Ticket), :Cabin) # hide
+```
+
+## ML.schema
+
+Schema is a programmatic description of the data in each column. It is a dictionary which maps each column (by name) to its schema type (mainly `Continuous`, and `Categorical`).
+
+- `ML.Continuous`: data is drawn from the real number line (e.g. Age)
+- `ML.Categorical`: data is drawn from a fixed set of values (e.g. Sex)
+
+`ML.schema(train_table)` will go through the data and infer the types and distribution of data. Let's try it without any arguments on the titanic dataset:
+
+```@example titanic
+ML.schema(train_table)
+```
+
+Here is how the schema was inferred:
+
+- Numeric fields were inferred to be `Continuous`, their mean and standard deviations were computed. This will later be used in normalizing the column in the feature matrix using the formula `((value - mean) / standard_deviation)`. This will bring all columns to the same "scale" making the training more effective.
+- Some string columns are inferred to be `Categorical` (e.g. Sex, Embarked) - this means that the column is a [PooledArray](https://github.com/JuliaComputing/PooledArrays.jl), and is drawn from a small "pool" of values. For example Sex is either "male" or "female"; Embarked is one of "Q", "S", "C" or ""
```

Comment:
So you consider that `PooledArray` should only be used to store categorical data? I thought that the original idea when `PooledDataArray` was deprecated was that we would use two types instead, with `CategoricalArray` being intended for categorical data, and `PooledArray` for any kind of data, without making it necessarily categorical.

Reply:
CategoricalArray is not used in JuliaDB. PooledArray happens to be returned by TextParse (I it does this for up to 5% cardinality after over 400 rows), so we deal with it.

The main idea here is that we need to pick _some_ defaults based on the column type in a deterministic way -- the user should be able to tell before running ML.schema what will be computed, so they can supply hints as necessary. PooledArray usually ends up being some kind of categorical or ordinal data.

Reply:
Yeah, but I think I would have used a different rule, e.g. `AbstractArray{<:Real}` variables are continuous, and others are categorical. That way you wouldn't have to special case `PooledArray`, `CategoricalArray` or any type.